### PR TITLE
Implemented _eval_nseries() for Hyper

### LINKED ISSUE #18193: AttributeError: integrate(1 / (1 + x**4)**(S(1)/4), [x, 0, oo])
```
>>> from sympy import *
>>> x = Symbol('x')
>>> integrate(1 / (1 + x**4)**(S(1)/4), [x, 0, oo])
Traceback (most recent call last):
  File "test.py", line 40, in <module>
    print(integrate(1 / (1 + x**4)**(S(1)/4), [x, 0, oo]))
  File "/home/eward/se/sympy/integrals/integrals.py", line 1522, in integrate
    return integral.doit(**doit_flags)
  File "/home/eward/se/sympy/integrals/integrals.py", line 669, in doit
    evalued = Add(*others)._eval_interval(x, a, b)
  File "/home/eward/se/sympy/core/expr.py", line 964, in _eval_interval
    B = _eval_endpoint(left=False)
  File "/home/eward/se/sympy/core/expr.py", line 949, in _eval_endpoint
    C = limit(self, x, c, "+" if left else "-")
  File "/home/eward/se/sympy/series/limits.py", line 71, in limit
    return Limit(e, z, z0, dir).doit(deep=False)
  File "/home/eward/se/sympy/series/limits.py", line 261, in doit
    r = gruntz(e, z, z0, dir)
  File "/home/eward/se/sympy/series/gruntz.py", line 671, in gruntz
    r = limitinf(e0, z)
  File "/home/eward/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/eward/se/sympy/series/gruntz.py", line 435, in limitinf
    c0, e0 = mrv_leadterm(e, x)
  File "/home/eward/se/sympy/core/cache.py", line 94, in wrapper
    retval = cfunc(*args, **kwargs)
  File "/home/eward/se/sympy/series/gruntz.py", line 524, in mrv_leadterm
    series = calculate_series(f, w, logx=logw)
  File "/home/eward/se/sympy/series/gruntz.py", line 477, in calculate_series
    for t in e.lseries(x, logx=logx):
  File "/home/eward/se/sympy/core/expr.py", line 2924, in yield_lseries
    for si in s:
  File "/home/eward/se/sympy/core/expr.py", line 3136, in _eval_lseries
    series = self._eval_nseries(x, n=n, logx=logx)
  File "/home/eward/se/sympy/core/mul.py", line 1749, in _eval_nseries
    terms = [t.nseries(x, n=n, logx=logx) for t in self.args]
  File "/home/eward/se/sympy/core/mul.py", line 1749, in <listcomp>
    terms = [t.nseries(x, n=n, logx=logx) for t in self.args]
  File "/home/eward/se/sympy/core/expr.py", line 3223, in nseries
    return self._eval_nseries(x, n=n, logx=logx)
  File "/home/eward/se/sympy/core/function.py", line 673, in _eval_nseries
    a = [t.compute_leading_term(x, logx=logx) for t in args]
  File "/home/eward/se/sympy/core/function.py", line 673, in <listcomp>
    a = [t.compute_leading_term(x, logx=logx) for t in args]
AttributeError: 'TupleArg' object has no attribute 'compute_leading_term'
```

It is trying to compute the `nseries` of a hypergeometric function:

```
>>> hyper((S(1)/4, S(1)/4), (S(5)/4,), exp_polar(I*pi)/x**4).nseries()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/eward/se/sympy/core/expr.py", line 3221, in nseries
    return self.series(x, x0, n, dir)
  File "/home/eward/se/sympy/core/expr.py", line 2870, in series
    rv = self.subs(x, xpos).series(xpos, x0, n, dir, logx=logx)
  File "/home/eward/se/sympy/core/expr.py", line 2877, in series
    s1 = self._eval_nseries(x, n=n, logx=logx)
  File "/home/eward/se/sympy/core/function.py", line 673, in _eval_nseries
    a = [t.compute_leading_term(x, logx=logx) for t in args]
  File "/home/eward/se/sympy/core/function.py", line 673, in <listcomp>
    a = [t.compute_leading_term(x, logx=logx) for t in args]
AttributeError: 'TupleArg' object has no attribute 'compute_leading_term'
```

### PR DESCRIPTION
**Pull request by dhruvmendiratta6**
#### References to other Issues or PRs

Fixes #18193 

#### Brief description of what is fixed or changed
In reference to Issue #18193 the function was implemented. Example of
the working code is as follows:

In [1]: from sympy import hyper

In [2]: from sympy.abc import x, y

In [3]: hyper((1, 2),(1, 2), x)._eval_nseries(x, 7, y)
Out[3]: 1 + x + x**2/2 + x**3/6 + x**4/24 + x**5/120 + x**6/720 + O(x**7)

In [4]: hyper((1, 2),(3, 4, 5), x)._eval_nseries(x, 7, y)
Out[4]: 1 + x/30 + x**2/1200 + x**3/63000 + x**4/4233600 + x**5/355622400 + x**6/36578304000 + O(x**7)

#### Release Notes

* functions
   * Added _eval_nseries() functionality to hyper

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This will need tests

**Comment by dhruvmendiratta6** [comment]
Actually the following test is still failing. I committed it anyway as the code is working and I couldn't figure out what was going wrong.
` File "c:\users\mendiratta\sympy\sympy\functions\special\tests\test_hyper.py", line 347, in test_limits
    O(k**6) # issue 6350
AssertionError
`

**Comment by jksuom** [comment]
There a a couple of things that should be changed in the code. In the failing issue, the argument of the function is `k**2` so that is what should appear in the series expansion. Also I think that `self` should be added to the `super()` call. Then it should be possible to fix the failing test.

File: sympy/functions/special/hyper.py [hunk 1]
```diff
@@ -220,6 +220,33 @@ def _eval_rewrite_as_Sum(self, ap, bq, z, **kwargs):
...
+        for i in range(n):
+            Rf1 = 1
+            Rf2 = 1
+            for a in ap:
+                Rf1 *= RisingFactorial(a, i)
+
+            for b in bq:
+                Rf2 *= RisingFactorial(b, i)
+
+            terms.append(((Rf1 / Rf2) * (x ** i)) / factorial(i))
```
Comment (jksuom) [inline sympy/functions/special/hyper.py]:
```suggestion
            terms.append(((Rf1/Rf2) * (arg**i)) / factorial(i))
```

File: sympy/functions/special/hyper.py [hunk 2]
```diff
@@ -220,6 +220,33 @@ def _eval_rewrite_as_Sum(self, ap, bq, z, **kwargs):
         return Piecewise((Sum(coeff * z**n / factorial(n), (n, 0, oo)),
                          self.convergence_statement), (self, True))
 
+    def _eval_nseries(self, x, n, logx):
+
+        from sympy.functions import factorial, RisingFactorial
+        from sympy import Order, Add
+
+        x0 = self.args[2].limit(x, 0)
```
Comment (jksuom) [inline sympy/functions/special/hyper.py]:
```suggestion
        arg = self.args[2]
        x0 = arg.limit(x, 0)
```

File: sympy/functions/special/hyper.py [hunk 3]
```diff
@@ -220,6 +220,33 @@ def _eval_rewrite_as_Sum(self, ap, bq, z, **kwargs):
...
+        ap = self.args[0]
+        bq = self.args[1]
+
+        if x0 != 0:
+            return super()._eval_nseries(x, n, logx)
```
Comment (jksuom) [inline sympy/functions/special/hyper.py]:
```suggestion
           return super()._eval_nseries(self, x, n, logx)
```

**Comment by jksuom** [comment]
Actually, `terms` should be computed with expansion of `arg` as a series in `x`, but that is not needed for the existing tests.

File: sympy/functions/special/hyper.py (hunk 3 above)
Comment (jksuom) [reply sympy/functions/special/hyper.py]: I'm not sure of this, though. The docs say that if the second argument of `super` is omitted, then the method will be unbound but I don't know what will happen if both arguments are omitted. However, there is an example that seems to indicate that the returned method will be bound to `self`. Perhaps that should be tested.

Comment (jksuom) [reply sympy/functions/special/hyper.py]: Tests seem to show that the method will be bound. So better leave this unchanged.

Comment (dhruvmendiratta6) [reply sympy/functions/special/hyper.py]: I think what you are saying is the way it was done in python 2 and in python 3 both the arguments can be left omitted.

**Comment by dhruvmendiratta6** [comment]
After removing `self` argument from `super()._eval_nseries()` call which was added by mistake there are some differences while calling the `_eval_nseries()` of `hyper` and `exp` which are as follows:

```
>>> hyper((1,2),(1,2), x)._eval_nseries(y,7,None)
hyper((1, 2), (1, 2), x) + O(y**7)

>>> exp(x)._eval_nseries(y,7,None)
exp(x)

>>> exp(x**2)._eval_nseries(x,7,None)
1 + x**2 + x**4/2 + x**6/6 + x**8/24 + x**10/120 + x**12/720 + O(x**14)

>>> hyper((1,2),(1,2), x**2)._eval_nseries(x,7,None)
1 + x**2 + x**4/2 + x**6/6 + O(x**7)
```
is the test failing due to the difference in second output. As what I understood was that maximum power of `x` can be of the order of `n` but that is not the case with `exp._eval_nseries()`

**Comment by jksuom** [comment]
Thanks, this is ready to be merged.

**Comment by dhruvmendiratta6** [comment]
Is there any other way apart from adding tests (which is already done) to improve the code coverage

**Pull request merged**